copy/copy.go

```diff
@@ -269,8 +273,11 @@ func Image(ctx context.Context, policyContext *signature.PolicyContext, destRef,
 		if len(options.EnsureCompressionVariantsExist) > 0 {
 			return nil, fmt.Errorf("EnsureCompressionVariantsExist is not implemented when not creating a multi-architecture image")
 		}
+		if (options.DestinationCtx == nil || options.DestinationCtx.CompressionFormat == nil) && options.ForceCompressionFormat {
+			return nil, fmt.Errorf("cannot use ForceCompressionFormat with undefined default compression format")
+		}
 		// The simple case: just copy a single image.
-		single, err := c.copySingleImage(ctx, c.unparsedToplevel, nil, copySingleImageOptions{requireCompressionFormatMatch: false})
+		single, err := c.copySingleImage(ctx, c.unparsedToplevel, nil, copySingleImageOptions{requireCompressionFormatMatch: c.options.ForceCompressionFormat})
```

Comment:
At 3 copies, please turn this into a single shared function.

Maybe just
```go
func shouldRequireCompressionFormatMatch(options *Options) (bool, error)
```
or perhaps (uglier?)
```go
func basicSingleImageOptions(options *Options) (copySingleImageOptions, error)
```
OTOH it’s very unclear what “basic” means — typically defaults and logic should happen directly in `copySingleImage`, this would be more of a `nonInstanceCopyCloneSingleImageOptions`, and that’s just _tedious_.